File: typescript/src/problems1_100/median-of-two-sorted-arrays.ts
class Problem {
  findMedianSortedArrays(nums1: number[], nums2: number[]): number {
    throw new Error(`Not implemented (${nums1}, ${nums2}).`);
  }
}

class Default extends Problem {
  findMedianSortedArrays(nums1: number[], nums2: number[]): number {
    const nums: number[] = nums1.concat(nums2);
    nums.sort();
    const center: number = Math.floor(nums.length / 2);
    if (nums.length % 2 === 1) {
      return nums[center];
    }
    return (nums[center - 1] + nums[center]) / 2;
  }
}

const SOLUTIONS = [new Default()];
export default SOLUTIONS;

File: python/src/problems1_100/median_of_two_sorted_arrays.py
from abc import ABC, abstractmethod
from typing import override


class Problem(ABC):
    @abstractmethod
    def find_median_sorted_arrays(self, nums1: list[int], nums2: list[int]) -> float:
        pass


class Default(Problem):
    @override
    def find_median_sorted_arrays(self, nums1: list[int], nums2: list[int]) -> float:
        nums: list[int] = nums1 + nums2
        nums.sort()
        center: int = len(nums) // 2
        if len(nums) % 2 == 1:
            return nums[center]
        return (nums[center - 1] + nums[center]) / 2


SOLUTIONS: list[Problem] = [Default()]
